app/assets/javascripts/govuk_publishing_components/components/table-filter.js

```diff
@@ -0,0 +1,44 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function TableFilter ($module) {
+    this.$module = $module
+    this.searchInput = $module.querySelector('input[name="filter"]')
+    this.tableRows = $module.querySelector('table').querySelectorAll('.govuk-table__body .govuk-table__row')
```

Comment:
Does this code need to consider whether any of these elements might not be found? I don't know if there's a scenario where they wouldn't, but for example if there was no `table` element then this line would error.

Reply:
I think that if this component is being rendered the elements will by definition all be present. Additionally I have not been able to find any equivalents in any existing components.